docs/ENTITY_RELATIONSHIP_DIAGRAM.md

```diff
@@ -114,3 +114,87 @@ PDC would then:
 * Collect the most recent ExternalFieldValues for each CanonicalField for that Applicant.
 
 It would use the ApplicationFieldValue set as the primary source, and the ExternalFieldValue set as a secondary source.
+
+## Examples
+### Registering an Application Schema
+
+```mermaid
+sequenceDiagram
+  actor Admin
+  participant API
+  participant Database
+  Admin ->>+ API : Here is a new application schema
+  API ->>+ Database : Register any new canonical fields
+  Database ->>- API : OK!
+  API ->>+ Database : Register the new application schema
+  Database ->>- API : OK!
+  API ->>- Admin :  OK!
+```
+
+New `Application Schemas` will have to be externally defined; some day maybe we will make a user interface that generates an `Application Schema` definition, but for the short term this will be manually written JSON (or YAML, or something else highly structured).  The schema will define the full set of `Application Schema Fields` along with the name of the `Canonical Field` to which the `Application Schema Fields` map.
+
+This might look something like this:
+
+```
+{
+  "fields": [
+    {
+      "name": "Applicant Name",
+      "type": "string",
+      "canonicalField": "applicantName",
+    },
+    {
+      "name": "Have you ever seen the Mona Lisa?",
+      "type": "boolean",
+      "canonicalField": "haveSeenMonaLisa",
+    }
+  ]
+}
+```
+
+The PDC API would then ingest that new schema document.  It would first register any `Canonical Fields` that did not already exist.  It would then register the `Application Schema` and `Application Schema Fields`, with field-level associations to the `Canonical Fields`.
+
+The system does not differentiate between "core" and "custom" fields.  Rather, there will be a set of `Canonical Fields` that are used by varying numbers of `Application Schemas`.  We will likely see that some `Canonical Fields` are used more often than others, and some are only used by a single `Application Schema`.  We might choose a subset of the `Canonical Fields` to highlight in our documentation and might call those "core" fields; that decision is not directly relevant to the schema.
```

Comment:
I would say "The database does not differentiate..." because it is possible that the APIs or the UI could potentially have a concept of "core" fields. It might also be helpful to add something like "One can think of 'core' fields as those canonical fields we would expect to see in most or all application schemas, across organizations, whereas a 'custom' field can be thought of as those fields that might only appear for one application schema or one organization." I don't know if that helps relate the idea to the DB completely effectively.

Reply:
Great call re the db not differentiating

Re the second suggestion I'd like going to avoid providing a heuristic for core fields at this stage; I'm worried it might cause folks to start down a path that isn't relevant to this convo (I don't want folks to start deliberating on what a core field is yet, since the idea is we can have that convo later when we know more)